.github/workflows/desktop.yml

```diff
@@ -60,106 +60,37 @@ jobs:
 
-      - name: Install vcpkg packages (windows)
-        if: steps.cache_vcpkg.outputs.cache-hit!='true' && startsWith(matrix.os, 'windows')
-        run: |
-          ${{ github.workspace }}/external/vcpkg/bootstrap-vcpkg.bat -disableMetrics
-          ${{ github.workspace }}/external/vcpkg/vcpkg install @${{env.VCPKG_RESPONSE_FILE}}
-
-      - name: Install vcpkg packages (mac and linux)
-        if: steps.cache_vcpkg.outputs.cache-hit!='true' && startsWith(matrix.os, 'windows')!='true'
-        run: |
-          ${{ github.workspace }}/external/vcpkg/bootstrap-vcpkg.sh -disableMetrics
-          ${{ github.workspace }}/external/vcpkg/vcpkg install @${{env.VCPKG_RESPONSE_FILE}}
+      - name: Cache ccache files
+        if: startsWith(matrix.os, 'ubuntu') || startsWith(matrix.os, 'macos')
+        id: cache_ccache
+        uses: actions/cache@v2
+        with:
+          path: ccache_dir
+          key: dev-test-ccache-${{ env.MATRIX_UNIQUE_NAME }}
 
       - name: Setup python
         uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python_version }}
           architecture: ${{ matrix.architecture }}
 
-      - name: Get pip cache dir
-        id: get_pip_cache_dir
+      - name: Install prerequisites(cpp and python dependencies)
         run: |
-          echo "::set-output name=cachedir::$(pip cache dir)"
-
-      - name: pip cache
-        id: cache_pip
-        uses: actions/cache@v2
-        with:
-          path: ${{ steps.get_pip_cache_dir.outputs.cachedir }}
-          key: dev-pip-${{ runner.os }}-${{ matrix.python_version }}-${{ matrix.architecture }}-${{ hashFiles(format('{0}', env.PIP_REQUIREMENTS_FILE)) }}
+          python scripts/install_prereqs.py --ccache --generator "${{ matrix.generator}}"
 
-      - name: Install pip packages
+      - name: CMake configure and build
         run: |
-          python -m pip install --upgrade pip
-          pip install -r ${{ env.PIP_REQUIREMENTS_FILE }}
-
-      - name: Cache ccache files
-        if: env.CCACHE_INSTALLED
-        id: cache_ccache
-        uses: actions/cache@v2
-        with:
-          path: ccache_dir
-          key: dev-test-ccache-${{ env.MATRIX_UNIQUE_NAME }}
-
-      - name: Configure 
-        shell: bash
-        run: |
-          mkdir build
-          cmake \
-            -S . \
-            -B build \
-            -G "${{ matrix.generator }}" \
-            -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} \
-            -DCMAKE_TOOLCHAIN_FILE=external/vcpkg/scripts/buildsystems/vcpkg.cmake \
-            -DVCPKG_TARGET_TRIPLET=${{ matrix.vcpkg_triplet }} \
-            -DFIREBASE_CPP_BUILD_TESTS=ON \
-            -DFIREBASE_INCLUDE_FIRESTORE=OFF \
-            -DFIREBASE_FORCE_FAKE_SECURE_STORAGE=ON
-
-      - name: Build
-        shell: bash
-        run: cmake --build build --config ${{ matrix.build_type }} -j 8
+          python scripts/build.py --build_tests
```

Comment:
Here is really where I'd like to see this be much more explicit. Something like...

build.py --platform=windows --arch=x64 --type=debug --runtime=md

(where obviously those come from the matrix, similar to how the ios build script is run in my PR).

Reply:
(because right now how does it even get those options?)

Reply:
Good catch. I forgot to add --arch=<architecture> and --config=<config> to this step. 
Its working right now because,
- platform is easy to detect in the script
- arch has a default value of 'x64' if its not specified (but we should keep it explicit here)
- type defaults to Debug in cmake (but we should specify it here)
- runtime (I havent gotten that far to support runtime in Desktop builds yet. Yes, we should have the runtime specified too)